src/app/vault/password-generator.component.html

```diff
@@ -62,148 +90,356 @@
+                    [disabled]="enforcedPasswordPolicyOptions?.useLowercase"
+                    [(ngModel)]="passwordOptions.lowercase"
+                  />
+                </div>
+                <div class="box-content-row box-content-row-checkbox" appBoxRow>
+                  <label for="numbers">0-9</label>
+                  <input
+                    id="numbers"
+                    type="checkbox"
+                    (change)="savePasswordOptions()"
+                    [disabled]="enforcedPasswordPolicyOptions?.useNumbers"
+                    [(ngModel)]="passwordOptions.number"
+                  />
+                </div>
+                <div class="box-content-row box-content-row-checkbox" appBoxRow>
+                  <label for="special">!@#$%^&*</label>
+                  <input
+                    id="special"
+                    type="checkbox"
+                    (change)="savePasswordOptions()"
+                    [disabled]="enforcedPasswordPolicyOptions?.useSpecial"
+                    [(ngModel)]="passwordOptions.special"
+                  />
+                </div>
+              </div>
+            </div>
+            <div class="box" [hidden]="!showOptions">
+              <div class="box-content condensed">
+                <div class="box-content-row box-content-row-input" appBoxRow>
+                  <label for="min-number">{{ "minNumbers" | i18n }}</label>
+                  <input
+                    id="min-number"
+                    type="number"
+                    min="0"
+                    max="9"
+                    (blur)="savePasswordOptions()"
+                    [(ngModel)]="passwordOptions.minNumber"
+                  />
+                </div>
+                <div class="box-content-row box-content-row-input" appBoxRow>
+                  <label for="min-special">{{ "minSpecial" | i18n }}</label>
+                  <input
+                    id="min-special"
+                    type="number"
+                    min="0"
+                    max="9"
+                    (blur)="savePasswordOptions()"
+                    [(ngModel)]="passwordOptions.minSpecial"
+                  />
+                </div>
+                <div class="box-content-row box-content-row-checkbox" appBoxRow>
+                  <label for="ambiguous">{{ "ambiguous" | i18n }}</label>
+                  <input
+                    id="ambiguous"
+                    type="checkbox"
+                    (change)="savePasswordOptions()"
+                    [(ngModel)]="avoidAmbiguous"
+                  />
+                </div>
+              </div>
+            </div>
+          </ng-container>
+        </ng-container>
+        <ng-container *ngIf="type === 'username'">
+          <div class="box">
+            <div class="box-header">
+              <button
+                type="button"
+                (click)="toggleOptions()"
+                appA11yTitle="{{ 'toggleVisibility' | i18n }}"
```

Comment:
Should expose state to a11y utilities:

```suggestion
                appA11yTitle="{{ 'toggleVisibility' | i18n }}"
                [attr.aria-expanded]="showOptions"
```